akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2Blueprint.scala

```diff
@@ -30,8 +31,13 @@ private[http2] final case class Http2SubStream(initialHeaders: ParsedHeadersFram
 
 object Http2Blueprint {
   // format: OFF
-  def serverStack(): BidiFlow[HttpResponse, ByteString, ByteString, HttpRequest, NotUsed] = {
-    httpLayer() atop
+  def serverStack(settings: ServerSettings, log: LoggingAdapter): BidiFlow[HttpResponse, ByteString, ByteString, HttpRequest, NotUsed] = {
+    val parserSettings = settings.parserSettings
+    val httpHeaderParser = HttpHeaderParser.prime(HttpHeaderParser.unprimed(parserSettings, log, { info =>
```

Comment:
Priming currently uses uppercase header strings. I will check whether the parser is case sensitive - and if it is - supply additional `primeHttp2` method